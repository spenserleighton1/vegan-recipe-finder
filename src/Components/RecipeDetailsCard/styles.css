.recipe-card {
  width: 30vw;
}

.save-recipe-btn {
  color: #3F493F;
  font-size: 18px;
  font-family: 'Cinzel', serif;
  background-color: transparent;
  border: none;
}

.save-recipe-btn:hover {
  color: darkgreen;
}

.directions-link {
  color: #3F493F;
  font-size: 18px;
  font-family: 'Cinzel', serif;
  text-decoration: none;

}