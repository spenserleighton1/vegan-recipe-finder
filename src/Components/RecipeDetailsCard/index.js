import React from 'react';
import './styles.css';
import PropTypes from 'prop-types';

const RecipeDetailsCard = ({ title, publisher, ingredients, linkUrl, id, addFavorite, authUser }) => {
  return (
    <div className='recipe-card'>
      <h2>{ title }</h2>
      <p>{ publisher }</p>
      <ul>
        { ingredients.map((ingredient, index) => <li key={ index }>{ ingredient }</li>)}
      </ul>
      <button  
        className='save-recipe-btn' 
        disabled={ !authUser }
        onClick={ () => { addFavorite(id); } }>Save Recipe
      </button>
      <a className='directions-link' href={ linkUrl }>Directions</a>
    </div>
  );
};

export default RecipeDetailsCard;

RecipeDetailsCard.propTypes = {
  title: PropTypes.string,
  publisher: PropTypes.string,
  ingredients: PropTypes.array,
  linkUrl: PropTypes.url,
  id: PropTypes.number,
  addFavorite: PropTypes.func,
  authUser: PropTypes.object
};