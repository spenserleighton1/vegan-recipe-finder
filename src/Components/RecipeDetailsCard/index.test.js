import React from 'react';
import { shallow } from 'enzyme';
import RecipeDetailsCard from './'

describe('RecipeDetailsCard', () => {
  it('should match the snapshot', () => {
    const recipe = {
      title: 'neat test',
      publisher: 'neat publisher',
      ingredients: ['neat ingredients'],
      linkUrl: 'www.com',
      id: 666,
    }
    const mockAddFavorite = jest.fn();
    const mockAuthUser = true

    const wrapper = shallow(<RecipeDetailsCard
      {...recipe}
      addFavorite={ mockAddFavorite }
      authUser={ mockAuthUser } />)

    expect(wrapper).toMatchSnapshot()
  })
  it('should call addFavorite when the button is clicked', () => {
     const recipe = {
      title: 'neat test',
      publisher: 'neat publisher',
      ingredients: ['neat ingredients'],
      linkUrl: 'www.com',
      id: 666,
    }
    const mockAddFavorite = jest.fn();
    const mockAuthUser = true

    const wrapper = shallow(<RecipeDetailsCard
      {...recipe}
      addFavorite={ mockAddFavorite }
      authUser={ mockAuthUser } />)

    wrapper.find('.favorite-btn').simulate('click')

    expect(mockAddFavorite).toHaveBeenCalled()
  })
})