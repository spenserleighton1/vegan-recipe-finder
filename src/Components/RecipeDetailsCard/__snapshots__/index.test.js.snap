// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetailsCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeDetailsCard
    addFavorite={[Function]}
    authUser={true}
    id={666}
    ingredients={
        Array [
            "neat ingredients",
          ]
    }
    linkUrl="www.com"
    publisher="neat publisher"
    title="neat test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          neat test
</h2>,
        <p>
          neat publisher
</p>,
        <ul>
          <li>
                    neat ingredients
          </li>
</ul>,
        <button
          className="save-recipe-btn"
          disabled={false}
          onClick={[Function]}
>
          Save Recipe
</button>,
        <a
          className="directions-link"
          href="www.com"
>
          Directions
</a>,
      ],
      "className": "recipe-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "neat test",
        },
        "ref": null,
        "rendered": "neat test",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "neat publisher",
        },
        "ref": null,
        "rendered": "neat publisher",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              neat ingredients
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "neat ingredients",
            },
            "ref": null,
            "rendered": "neat ingredients",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Save Recipe",
          "className": "save-recipe-btn",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Save Recipe",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Directions",
          "className": "directions-link",
          "href": "www.com",
        },
        "ref": null,
        "rendered": "Directions",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            neat test
</h2>,
          <p>
            neat publisher
</p>,
          <ul>
            <li>
                        neat ingredients
            </li>
</ul>,
          <button
            className="save-recipe-btn"
            disabled={false}
            onClick={[Function]}
>
            Save Recipe
</button>,
          <a
            className="directions-link"
            href="www.com"
>
            Directions
</a>,
        ],
        "className": "recipe-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "neat test",
          },
          "ref": null,
          "rendered": "neat test",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "neat publisher",
          },
          "ref": null,
          "rendered": "neat publisher",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                neat ingredients
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "neat ingredients",
              },
              "ref": null,
              "rendered": "neat ingredients",
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Save Recipe",
            "className": "save-recipe-btn",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Save Recipe",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Directions",
            "className": "directions-link",
            "href": "www.com",
          },
          "ref": null,
          "rendered": "Directions",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
